app/models/sale.rb

```diff
@@ -337,6 +338,10 @@ def opened_financial_year?
     FinancialYear.on(invoiced_at)&.opened?
   end
 
+  def financial_year_in_closure_preparation?
```

Comment:
I think we can find a better name for this one, make it clear how the financial year is found. Maybe `invoiced_during_closure_preparation?` ?